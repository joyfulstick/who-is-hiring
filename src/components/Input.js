import React from 'react'
import LabelStyle from '../styles/LabelStyle'

const Input = ({ name, type, min, max, value, checked, handleChange }) => (
  <LabelStyle>
    {name}
    {type === 'range' && value}
    <input
      name={name}
      type={type}
      min={min}
      max={max}
      value={value}
      checked={checked}
      onChange={handleChange}
    />
  </LabelStyle>
)

export default Input
