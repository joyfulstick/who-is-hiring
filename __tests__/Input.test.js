import { shallow } from 'enzyme'
import React from 'react'
import Input from '../src/components/Input'

describe('<Input />', () => {
  it('renders correctly - type checkbox', () => {
    const wrapper = shallow(
      <Input
        name="isRemote"
        type="checkbox"
        checked={true}
        handleChange={jest.fn()}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly - type range', () => {
    const wrapper = shallow(
      <Input
        name="salary"
        type="range"
        min={0}
        max={20}
        value={5}
        handleChange={jest.fn()}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
