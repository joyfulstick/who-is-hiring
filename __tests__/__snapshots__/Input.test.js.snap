// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly - type checkbox 1`] = `
<styled.label>
  <span>
    isRemote
  </span>
  <input
    checked={true}
    name="isRemote"
    onChange={[MockFunction]}
    type="checkbox"
  />
</styled.label>
`;

exports[`<Input /> renders correctly - type range 1`] = `
<styled.label>
  <span>
    salary
  </span>
  <input
    max={20}
    min={0}
    name="salary"
    onChange={[MockFunction]}
    type="range"
    value={5}
  />
  <span
    className="amount"
  >
    5
  </span>
</styled.label>
`;
